src/beacon-api/app/api/v1/devices.py

```
@@ -223,6 +223,34 @@ async def get_device_files(device_id: str):
+        return device_service.get_synced_device_details(
+            db,
+            skip=skip,
+            limit=limit,
+            search=search,
+            group_device_ids=group_device_ids,
+        )
+    except HTTPException:
+        raise
+    except Exception as e:
+        logger.exception(f"Unexpected error fetching synced devices: {e}")
+        raise HTTPException(status_code=500, detail="Internal Server Error")
```

Ruff (0.15.12)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)